docs/VLSI/Sky130-Tutorial.rst

```diff
@@ -0,0 +1,157 @@
+
+  * Integration of Hammer's build system into Chipyard and abstracts away some Hammer commands.
+
+* build
+
+  * Hammer output directory. Can be changed with the ``OBJ_DIR`` variable.
+  * Will contain subdirectories such as ``syn-rundir`` and ``par-rundir`` and the ``inputs.yml`` denoting the top module and input Verilog files.
+
+* env.yml
+
+  * A template file for tool environment configuration. Fill in the install and license server paths for your environment.
+
+* example-vlsi-sky130
+
+  * Entry point to Hammer. Contains example placeholders for hooks.
+
+* example-sky130.yml, example-tools.yml
+
+  * Hammer IR for this tutorial.
+
+* example-design.yml, example-nangate45.yml, example-tech.yml
+
+  * Hammer IR not used for this tutorial but provided as templates.
+
+* generated-src
+
+  * All of the elaborated Chisel and FIRRTL.
+
+* hammer, hammer-<vendor>-plugins, hammer-<tech>-plugin
+
+  * Core, tool, tech repositories.
+
+Prerequisites
+-------------
+
+* Python 3.4+
+* numpy package
+* Genus, Innovus, and Calibre licenses
+* Sky130 PDK, install using `these directions  <https://github.com/ucb-bar/hammer/blob/master/src/hammer-vlsi/technology/sky130/README.md>`__
+
+Initial Setup
+-------------
+In the Chipyard root, run:
+
+.. code-block:: shell
+
+    ./scripts/init-vlsi.sh sky130
+    
+to pull the Hammer & plugin submodules. Note that for technologies other than ``sky130`` or ``asap7``, the tech submodule must be added in the ``vlsi`` folder first.
+
+Pull the Hammer environment into the shell:
+
+.. code-block:: shell
+
+    cd vlsi
+    export HAMMER_HOME=$PWD/hammer
+    source $HAMMER_HOME/sourceme.sh
+
+Building the Design
+--------------------
+To elaborate the ``TinyRocketConfig`` and set up all prerequisites for the build system to push the design and SRAM macros through the flow:
+
+.. code-block:: shell
+
+    make buildfile tech_name=sky130 CONFIG=TinyRocketConfig
+
+The ``CONFIG=TinyRocketConfig`` selects the target generator config in the same manner as the rest of the Chipyard framework. This elaborates a stripped-down Rocket Chip in the interest of minimizing tool runtime.
+
+For the curious, ``make buildfile`` generates a set of Make targets in ``build/hammer.d``. It needs to be re-run if environment variables are changed. It is recommended that you edit these variables directly in the Makefile rather than exporting them to your shell environment.
+
+Running the VLSI Flow
+---------------------
+
+example-vlsi-sky130
+^^^^^^^^^^^^^^^^^^^
+This is the entry script with placeholders for hooks. In the ``ExampleDriver`` class, a list of hooks is passed in the ``get_extra_par_hooks``. Hooks are additional snippets of python and TCL (via ``x.append()``) to extend the Hammer APIs. Hooks can be inserted using the ``make_pre/post/replacement_hook`` methods as shown in this example. Refer to the Hammer documentation on hooks for a detailed description of how these are injected into the VLSI flow.
```

Comment:
I would move the "In the ExampleDriver" text to before the "Refer to the Hammer" sentence.